tools/flash_shelly.py

```diff
@@ -96,16 +96,45 @@
   pipe = subprocess.check_output(['pip3', 'install', 'requests'])
   import requests
 
+def get_info(host):
+  if not '.local' in host:
+    host = host + '.local'
+  try:
+    with urllib.request.urlopen(f'http://{host}/rpc/Shelly.GetInfo') as fp:
+      info = json.load(fp)
+    info['fw_type'] = 'homekit'
+    if not 'device_id' in info:
+      info['device_id'] = info['id']
+    return((info['host'], info))
+  except (urllib.error.URLError) as err:
+    if 'Errno 8' in str(err.reason):
+      logger.warning(f"{RED}Could not resolve host: {host}\n{NC}")
+      return(('None', 'None'))
```

Comment:
why double parens around return?

also, it should be `None` without quotes - None is a null value in python

Reply:
because of this:

```
Traceback (most recent call last):
  File "./flash_shelly.py", line 506, in <module>
    device_scan(args.hosts, action, args.do_all, args.dry_run, args.silent_run, args.mode, args.exclude, args.version, args.variant, stock_release_info, homekit_release_info)
  File "./flash_shelly.py", line 435, in device_scan
    parse_info(device[1], action, dry_run, silent_run, mode, exclude, version, variant, stock_release_info, homekit_release_info)
  File "./flash_shelly.py", line 279, in parse_info
    cfw = device_info['version']       # current firmware version
TypeError: 'NoneType' object is not subscriptable
```

Reply:
i'll see if i can look again.

Reply:
and 
```
Traceback (most recent call last):
  File "./flash_shelly.py", line 506, in <module>
    device_scan(args.hosts, action, args.do_all, args.dry_run, args.silent_run, args.mode, args.exclude, args.version, args.variant, stock_release_info, homekit_release_info)
  File "./flash_shelly.py", line 430, in device_scan
    device_list.sort()
TypeError: '<' not supported between instances of 'str' and 'NoneType'
```

Reply:
> why double parens around return?

it has to be a list, not 2 strings.

Reply:
this is a normal return

```
device_test: [('Andys-Bedroom-SW.local', {'device_id': 'shellyswitch25-05B323', 'name': 'Andys-Bedroom-SW', 'app': 'switch25', 'model': 'Shelly25', 'stock_model': 'SHSW-25', 'host': 'Andys-Bedroom-SW.local', 'version': '2.3.0', 'fw_build': '20201012-201234/2.3.0-ga36048b-master', 'uptime': 290593, 'wifi_en': True, 'wifi_ssid': 'ABFi-IoT-2.4', 'wifi_pass': '***', 'wifi_rssi': -53, 'wifi_ip': '192.168.3.48', 'hap_cn': 10, 'hap_running': True, 'hap_paired': True, 'hap_ip_conns_pending': 0, 'hap_ip_conns_active': 8, 'hap_ip_conns_max': 9, 'sys_mode': 0, 'rsh_avail': True, 'gdo_avail': True, 'debug_en': False, 'components': [{'id': 1, 'type': 0, 'name': 'Lights', 'svc_type': 0, 'in_mode': 0, 'initial': 2, 'state': False, 'auto_off': False, 'auto_off_delay': 10.0, 'apower': 0.0, 'aenergy': 5.36}, {'id': 2, 'type': 0, 'name': 'Andys Bedroom SW2', 'svc_type': -1, 'in_mode': 3, 'initial': 2, 'state': False, 'auto_off': False, 'auto_off_delay': 10.0, 'apower': 0.0, 'aenergy': 0.0}, {'id': 2, 'type': 3, 'name': 'Shades SW', 'in_mode': 0, 'last_ev': 0, 'last_ev_age': 37196.953}], 'fw_type': 'homekit'}), ('Bathroom-SW.local', {'id':..........
```

Reply:
yeah `None` has to be a string, you can not sort list with None type.

Reply:
that's because you have to be careful and not mix None with other types.
sure, `'None`' is subscriptable, but you're effectively covering up the fact that you're trying to use it as a dict. it's smelly code.
test for None-ness (`if x is None: ...`), bail out or skip that entry if necessary

Reply:
i have, but you can't sort on none type.

Reply:
let me try to remove the None from list before we use it.

Reply:
ok hows this https://github.com/mongoose-os-apps/shelly-homekit/pull/198/commits/3284c4a019a1ff19900c6ee40bd9dc8336ff0f66